src/openrct2/world/Map.cpp

```diff
@@ -3185,25 +3136,30 @@ void map_reorganise_elements()
  *  Returns true on space available for more elements
  *  Reorganises the map elements to check for space
  */
-bool map_check_free_elements_and_reorganise(sint32 num_elements)
+bool map_check_free_elements_and_reorganise(sint32 numElements)
 {
-    if ((gNextFreeTileElement + num_elements) <= gTileElements + MAX_TILE_ELEMENTS)
-        return true;
-
-    for (sint32 i = 1000; i != 0; --i)
-        sub_68B089();
-
-    if ((gNextFreeTileElement + num_elements) <= gTileElements + MAX_TILE_ELEMENTS)
-        return true;
+    if (numElements != 0)
+    {
+        auto tileElementEnd = &gTileElements[MAX_TILE_ELEMENTS];
```

Comment:
I think the index here needs to be `MAX_TILE_ELEMENTS - 1`, or the check below should be `>=` instead.